### PR TITLE
fix: use impl Into<Cookie>

### PR DESCRIPTION
**Pull request by mseele**
closes https://github.com/tokio-rs/axum/issues/2347

### CODE REVIEW

File: axum-extra/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning].
 - **added:** `Clone` implementation for `ErasedJson` ([#2142])
 - **breaking:** Update to prost 0.12. Used for the `Protobuf` extractor
 - **breaking:** Make `tokio` an optional dependency
+- **breaking**: Functions and methods that previously accepted a `Cookie` now accept any `T: Into<Cookie>` ([#2347])
```
Comment (jplatte) [inline axum-extra/CHANGELOG.md:15]:
```suggestion
- **breaking**: Functions and methods that previously accepted a `Cookie` now accept any `T: Into<Cookie>` ([#2348])
```

Also needs a link definition below.

**Review by jplatte — COMMENTED** [review]
Thanks for the PR!

This only modifies `CookieJar`, but I think `PrivateCookieJar` and `SignedCookieJar` have the same methods, right?

Finally a small note, if you start working on a help-wanted issue, I would recommend leaving a comment on the issue first to let others know you're working on it, to avoid duplicate work.

File: axum-extra/CHANGELOG.md (hunk 1 above)
Comment (jplatte) [reply axum-extra/CHANGELOG.md]: The link definition is still missing.

**Comment by mseele** [comment]
Thanks for pointing out that `PrivateCookieJar` and `SignedCookieJar` should also be refactored, I've updated them.

And sorry for not noting that I'm working on a PR. Just saw the issue and jumped right in. Will to that next time 👍

File: axum-extra/CHANGELOG.md (hunk 1 above)
Comment (mseele) [reply axum-extra/CHANGELOG.md]: Ah, I see. You'll have that in a minute.

**Review by jplatte — APPROVED** [review]

**Pull request merged**